devito/types/dense.py

```diff
@@ -94,7 +94,7 @@ def __init_finalize__(self, *args, function=None, **kwargs):
             # a reference to the user-provided buffer
             self._initializer = None
             if len(initializer) > 0:
-                self.data_with_halo[:] = initializer
+                self.data_with_halo[:] = initializer[:]
```

Comment:
interesting, what's the difference here?

Reply:
No difference, just debug leftover, the issue was calling `__init_finalize__` before setting `newobj.function` above, needed for first touch init. Slightly safer in case of shape differences though for weird `(1, n)/(n, 1)` shape where someone might give and `(n,)` vector

Reply:
`[:]` is just a view it has no performance impact.

Reply:
well the view gets created, so it's not really 0 impact.... but yeah, I agree it's gonna be 0.0000001 in practice 😬